pkg/tsdb/cloudwatch/cloudwatch_test.go

```diff
@@ -72,3 +84,114 @@ func TestNewInstanceSettings(t *testing.T) {
+		client = FakeCheckHealthClient{
+			describeLogGroupsWithContext: func(ctx aws.Context, input *cloudwatchlogs.DescribeLogGroupsInput,
+				options ...awsrequest.Option) (*cloudwatchlogs.DescribeLogGroupsOutput, error) {
+				return nil, fmt.Errorf("some logs query error")
+			}}
+		im := datasource.NewInstanceManager(func(s backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+			return datasourceInfo{}, nil
+		})
+
+		executor := newExecutor(im, newTestConfig(), fakeSessionCache{})
+		request := &backend.CheckHealthRequest{
+			PluginContext: backend.PluginContext{
+				DataSourceInstanceSettings: &backend.DataSourceInstanceSettings{},
+			},
+		}
+		resp, err := executor.CheckHealth(context.Background(), request)
+		require.NoError(t, err)
+
+		assert.Equal(t, &backend.CheckHealthResult{
+			Status:  backend.HealthStatusError,
+			Message: "1. Successfully queried the CloudWatch metrics API.\n2. CloudWatch logs query failed: some logs query error",
+		}, resp)
+	})
+
+	t.Run("successfully queries logs, fails during metrics query", func(t *testing.T) {
+		client = FakeCheckHealthClient{
+			listMetricsPages: func(input *cloudwatch.ListMetricsInput, fn func(*cloudwatch.ListMetricsOutput, bool) bool) error {
+				return fmt.Errorf("some list metrics error")
+			}}
+		im := datasource.NewInstanceManager(func(s backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+			return datasourceInfo{}, nil
+		})
+
+		executor := newExecutor(im, newTestConfig(), fakeSessionCache{})
+		request := &backend.CheckHealthRequest{
+			PluginContext: backend.PluginContext{
+				DataSourceInstanceSettings: &backend.DataSourceInstanceSettings{},
+			},
+		}
+		resp, err := executor.CheckHealth(context.Background(), request)
+		require.NoError(t, err)
+
+		assert.Equal(t, &backend.CheckHealthResult{
+			Status:  backend.HealthStatusError,
+			Message: "1. CloudWatch metrics query failed: some list metrics error\n2. Successfully queried the CloudWatch logs API.",
+		}, resp)
+	})
+
+	t.Run("fail to get clients", func(t *testing.T) {
+		client = FakeCheckHealthClient{}
+		im := datasource.NewInstanceManager(func(s backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+			return datasourceInfo{}, nil
+		})
+
+		sessionsCli := fakeSessionCache{getSession: func(c awsds.SessionConfig) (*session.Session, error) {
+			return nil, fmt.Errorf("some sessions error")
+		}}
+		executor := newExecutor(im, newTestConfig(), sessionsCli)
+		request := &backend.CheckHealthRequest{
+			PluginContext: backend.PluginContext{
+				DataSourceInstanceSettings: &backend.DataSourceInstanceSettings{},
+			},
+		}
+		resp, err := executor.CheckHealth(context.Background(), request)
+		require.NoError(t, err)
+
+		assert.Equal(t, &backend.CheckHealthResult{
+			Status:  backend.HealthStatusError,
+			Message: "1. CloudWatch metrics query failed: some sessions error\n2. CloudWatch logs query failed: some sessions error",
+		}, resp)
+	})
+}
```

Comment:
there is not a lot of shared code but do you think it's worth it to move these tests to a test array?

Reply:
done

Reply:
Sorry to jump in later!
I don't think it's worth it to make this a test array. I think it's a little more jumping back and forth to read, make some tests a bit busy (introduces some fields that some tests don't explicitly need defined).
If no one is strongly opposed, I'd rather leave it as it was. Hope that would be alright.

Reply:
yeah, not feeling strong about this one. I usually prefer to use table tests because it's easier to maintain tests in the future but here there are not much boilerplate code so up to you.